app/imports/api/photosMethods.js

```diff
@@ -21,31 +21,36 @@ export const addPost = new ValidatedMethod({
   }).validator(),
 
   run({photo, fingerprint, location, caption}) {
-
-    let hashtagsArray;
-
-    if (caption !== undefined) {
-
-      // Sanitize caption
-      caption = sanitizeHtml(caption);
-
-      // Parse hastags
-      hashtagsArray = caption.match(/#\S+/g);
-    }
-
     let emptyVotes = [];
 
     let data = {
       photo,
       fingerprint,
       location,
-      caption,
       upVotes: emptyVotes,
       downVotes: emptyVotes,
     }
 
-    if (hashtagsArray) {
-      data.hashtags = hashtagsArray;
+    // If caption is present sanitize it and parse for hashtags
+    if (caption !== undefined) {
+
+      // Sanitize caption
+      let sanitizedCaption = sanitizeHtml(caption);
+
+      // Add to `data`
+      data.caption = sanitizedCaption;
+
+      // Parse hastags
+      const hashtagsArray = _.compact(sanitizedCaption.match(/#\S+/g));
+
+      // lowercase the hastags
+      hashtagsArrayLower = _.map(hashtagsArray, hashtag => hashtag.toLowerCase());
```

Comment:
missing `var` declaration. `hashtagsArray` could be a `let` and have its value overwritten here. Same for above in comments method